EC2 non-Docker deployment using 'nightly' Impostor build

mrjohnson1024 opened the pull request:
We'll still need to figure out the config.json if this is going to run with a cert behind a load balancer. I don't have a way to test that currently since I'm using my home vSphere cluster.

mrjohnson1024 committed:
Docker Impostor on EC2 Amazon Linux

mrjohnson1024 committed:
added additional playbook for running docker-compose

mrjohnson1024 committed:
added additional playbook for running docker-compose

mrjohnson1024 committed:
updated playbook for RHEL; added recreate to docker_compose

mrjohnson1024 committed:
fixed filename to be consistent

mrjohnson1024 committed:
Merge branch 'rayprogramming:main' into ec2_playbook

mrjohnson1024 committed:
cleaned up dir

mrjohnson1024 committed:
Merge branch 'ec2_playbook' of https://github.com/mrjohnson1024/terraform-aws-impostor-server into ec2_playbook

mrjohnson1024 committed:
updated working non-Docker version for nightly build

Rihoj commented on ansible/playbook.yml:
```diff
@@ -0,0 +1,163 @@
+          state: latest
+          update_cache: true
+
+      - name: Get our external IP address to set in config.json
```
We should no need to pull the external config and supply a domain name. In my PR I am going to be passing in domain values to allow ansible to set the variable. It will be the game_domain.
https://github.com/rayprogramming/terraform-aws-impostor-server/pull/11/files#diff-6c27bcefa33af3496292d0448a95189bc1f75e38a9a6c9970e4490bef4c98a84R45

mrjohnson1024 replied:
Yeah I figured we'd need to pull in the LB IP info with TF but I wasn't sure, this was more for running it in my own homelab but I can remove that extra stuff if you don't need it.

Rihoj commented on ansible/playbook.yml:
```diff
@@ -0,0 +1,163 @@
+    impostor_path: "/home/ec2-user/Impostor"
+    impostor_plugins: "/home/ec2-user/Impostor/Plugins"
+    impostor_user: "ec2-user"
+    public_port: "443"
```
```
"Server": {
    "PublicIp": "34.232.46.194",
    "PublicPort": 22023,
    "ListenIp": "0.0.0.0",
    "ListenPort": 22023
  },
```

That is my current config. The Public IP will need changed to a domain for a NLB, but then the public and listen ports will need to be 22023. This is because during match making the Impostor.Http will read the PublicIP and port as what UDP endpoint to give to the client to tell it where to connect to.

mrjohnson1024 replied:
Should we just hardcode your public IP or FQDN into our config, or did we want to keep the playbook more generic so it could be used elsewhere?

mrjohnson1024 replied:
Actually, looking over your other PR, I see you are setting "'matchmaker_domain=${local.matchmaker_fqdn}' -e 'game_domain=${local.gameserver_fqdn}'" in the Ansible provisioner. Which one of these goes where? I'm not 100% yet on what exactly the matchmaker is doing or how it works. I assume "PublicIP" in the config.json should be "gameserver_fqdn", right?

Rihoj commented on ansible/configs/config_http.json:
```diff
@@ -0,0 +1,8 @@
+{
+    "HttpServer": {
+      "ListenIp": "{{ ansible_host }}",
```
I don't see ansible_host defined anywhere. However, this should be 0.0.0.0

mrjohnson1024 replied:
"ansible_host" is a magic variable: https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_vars_facts.html#information-about-ansible-magic-variables

It's just the SSH IP address that Ansible is using for that particular host. 0.0.0.0 should work fine too but I wasn't sure, some apps are different but that should bind it to any available IPv4 address, so I can make that change.

Rihoj commented on ansible/configs/config.json:
```diff
@@ -0,0 +1,24 @@
+    },
+    "Debug": {
+      "GameRecorderEnabled": true,
+      "GameRecorderPath": "{{ impostor_path }}/records/"
```
We may need to move this later. I am attaching a [whole drive](https://github.com/rayprogramming/terraform-aws-impostor-server/pull/11/files#diff-6c27bcefa33af3496292d0448a95189bc1f75e38a9a6c9970e4490bef4c98a84R57) to hold records.

mrjohnson1024 replied:
Okay no problem, we can just pass that var into the provisioner when the playbook is run and use that instead

Rihoj reviewed (changes requested):
This is looking really good and look like it will fit great.

I do think we should make the target upstream branch be getting_started though.

mrjohnson1024 committed:
Update config.json

set listenip to quad 0

mrjohnson1024 committed:
Update config.json

mrjohnson1024 committed:
added listen_ip var to push via templates

in case we want to change the private IP info later

mrjohnson1024 committed:
Update config_http.json

added listen_ip

mrjohnson1024 committed:
Update playbook.yml

removed curl for public IP info (not needed since we are using a load balancer)

Rihoj commented:
Oh wow, I didn't realize that closing the parent branch auto closed this one :(

mrjohnson1024 commented:
Should I create a new branch, or do you want me to just redo the PR and point it to "main"?

Rihoj commented:
new PR point it to main please.
